### PR TITLE
Fixes: Topic play new implementation

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

This PR changes the implementation of TopicPlay recyclerview.

https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/94a5f38c-e475-4514-8784-0e42cf70c25e/TP-Play-Scrolled-Up-

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/topic/play/StorySummaryAdapter.kt [hunk 1]
```diff
@@ -2,85 +2,126 @@ package org.oppia.app.topic.play
...
 import org.oppia.app.databinding.TopicPlayStorySummaryBinding
+import org.oppia.app.databinding.TopicPlayTitleBinding
 import org.oppia.app.model.ChapterPlayState
 import org.oppia.app.model.ChapterSummary
-import org.oppia.app.model.StorySummary
 
 // TODO(#216): Make use of generic data-binding-enabled RecyclerView adapter.
 
+private const val VIEW_TYPE_TITLE_TEXT = 1
+private const val VIEW_TYPE_STORY_ITEM = 2
```
Comment (veena14cs) [inline app/src/main/java/org/oppia/app/topic/play/StorySummaryAdapter.kt]: These 2 items doesn't seem to be used anywhere.

File: app/src/main/java/org/oppia/app/topic/play/StorySummaryViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package org.oppia.app.topic.play
+
+import android.view.View
+import androidx.lifecycle.ViewModel
+import org.oppia.app.model.PromotedStory
```
Comment (veena14cs) [inline app/src/main/java/org/oppia/app/topic/play/StorySummaryViewModel.kt]: nit: Remove unused import.

**Review by veena14cs — APPROVED** [review]
Looks good.

File: app/src/main/java/org/oppia/app/topic/play/StorySummaryViewModel.kt (hunk 2 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/topic/play/StorySummaryViewModel.kt]: Done.

**Comment by rt4914** [comment]
Merging this as this is a very similar implementation to StateFragment, HomeFragment, ContinuePlaying, etc.

**Pull request merged**